### PR TITLE
`Polymer.dom(element).observeNodes(callback)`: ensure callback is cal…

### PR DESCRIPTION
**Pull request by sorvell**

### CODE REVIEW

File: src/utils/polymer.dom.html [hunk 1]
```diff
@@ -244,7 +244,7 @@
       let list = [];
       for (let i=0, l=c$.length, c; (i<l) && (c=c$[i]); i++) {
         if ((c.nodeType === Node.ELEMENT_NODE) &&
-            Polymer.Utils.matchesSelector(this.node, selector)) {
+            Polymer.Utils.matchesSelector(c, selector)) {
```
Comment (bicknellr) [inline src/utils/polymer.dom.html]: If you're going down the same rabbit hole as me right now, I think https://github.com/Polymer/polymer/blob/fd052da7f8e4aa7432a9ed36e55345d77e2aa0f3/src/utils/polymer.dom.html#L24 is going to end up being your next change. :) `node.childNodes` -> `(node.root ? node.root : node).childNodes` or something similar.

Comment (sorvell) [inline src/utils/polymer.dom.html]: I believe we shouldn't need to check `.root` there unless there is a Shady DOM bug. Let's deal with that separately if so.

Comment (bicknellr) [inline src/utils/polymer.dom.html]: Ok, I'll open a bug.

Comment (bicknellr) [inline src/utils/polymer.dom.html]: Polymer/polymer#4000

**Pull request merged**